lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,398 @@
+                             gettext('Are you sure you want to select this session?') :
+                             gettext('Are you sure you want to change to a different session?');
+                         confirmationMsgBody = !currentSessionId ? '' :
+                             gettext('Any course progress or grades from your current session will be lost.');
+                     } else {
+                         confirmationMsgTitle = gettext('Are you sure that you would like to leave this session?');
+                         confirmationMsgBody = gettext('Any course progress or grades from your current session will ' +
+                                'be lost.');
+                     }
+
+                     // Remove existing popover and re-initialize
+                     popoverDialogHtml = this.verificationTpl({
+                         confirmationMsgTitle: confirmationMsgTitle,
+                         confirmationMsgBody: confirmationMsgBody
+                     });
+
+                     invokingElement.popover({
+                         placement: 'bottom',
+                         container: this.$el,
+                         html: true,
+                         trigger: 'click',
+                         content: popoverDialogHtml.text
+                     });
+                 },
+
+                 removeDialog: function(invokingElement) {
+                     /* Removes the Bootstrap v4 dialog modal from the update session enrollment button. */
+                     invokingElement.popover('dispose');
+                 },
+
+                 showDialog: function(invokingElement) {
+                     /* Given an element with an associated dialog modal, shows the modal. */
+                     invokingElement.popover('show');
+                     this.$('.final-confirmation-btn:first').focus();
+                 },
+
+                 hideDialog: function(el, returnFocus) {
+                     /* Hides the  modal without removing it from the DOM. */
+                     el = el instanceof jQuery ? el : this.$('.enroll-btn-initial');
+                     el.popover('hide');
+                     if (returnFocus) {
+                         el.focus();
+                     }
+                 },
+
+                 handleVerificationPopoverA11y: function(e) {
+                     /* Ensure that the second step verification popover is treated as an a11y compliant dialog */
+                     var nextButton,
+                         openButton = $(e.target).closest('.course-entitlement-selection-container')
+                             .find('.enroll-btn-initial');
+                     if (e.key === 'Tab') {
+                         e.preventDefault();
+                         nextButton = $(e.target).is(':first-child') ? $(e.target).next('.final-confirmation-btn') :
+                             $(e.target).prev('.final-confirmation-btn');
```

Comment:
Could we DRY this up by saving `$(e.target)` to a variable?

Reply:
Yeah sure